controllers/azurecluster_reconciler.go

```diff
@@ -138,7 +141,41 @@ func (r *azureClusterReconciler) Delete(ctx context.Context) error {
 			return errors.Wrapf(err, "failed to delete resource group")
 		}
 	}
+	if r.scope.AzureCluster.Spec.IdentityName != nil {
+		if err := r.deleteAzureIdentity(ctx); err != nil {
+			return errors.Wrapf(err, "failed to delete azure identity")
+		}
+	}
+
+	return nil
+}
 
+func (r *azureClusterReconciler) deleteAzureIdentity(ctx context.Context) error {
+	identityNamesToDelete := []string{}
+	var bindings aadpodv1.AzureIdentityBindingList
+	if err := r.scope.Client.List(ctx, &bindings, client.InNamespace("capz-system")); err != nil {
+		return errors.Wrapf(err, "failed to list AzureIdentityBinding")
+	}
+	for _, b := range bindings.Items {
+		binding := b
+		if binding.ObjectMeta.Labels[clusterv1.ClusterLabelName] == r.scope.AzureCluster.Name &&
+			binding.ObjectMeta.Labels[infrav1.ClusterLabelNamespace] == r.scope.AzureCluster.Namespace {
+
+			identityNamesToDelete = append(identityNamesToDelete, binding.Spec.AzureIdentity)
+			if err := r.scope.Client.Delete(ctx, &binding); err != nil {
```

Comment:
I might be misreading the code, but, are we deleting the identity that wasn't created by CAPZ? or is it created somewhere else? 

If it's applied as part of the deployment, would it make sense to have this in a separate reconciler since it's not technically part of the AzureCluster resource?

Reply:
so we create a copy of the azureidentity in the capz-system namespace so the NMI pod can use it, and that's the one we delete. its created as part of the initialization of the cluster scope 

do you think it needs a different reconciler to remove it if the cluster is deleted? I had that at some point, but removed it

Reply:
we now have azureidentity_controller